pkg/propertyprovider/azure/cloudconfig/config.go

```diff
@@ -0,0 +1,155 @@
+package cloudconfig
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient/policy/ratelimit"
+	"sigs.k8s.io/cloud-provider-azure/pkg/consts"
+)
+
+const (
+	// DefaultUserAgent is the default user agent string to access Azure resources.
+	DefaultUserAgent = "fleet-member-agent"
+)
+
+// CloudConfig holds the configuration parsed from the --cloud-config flag
+type RateLimitConfig ratelimit.Config
+type CloudConfig struct {
+	azclient.ARMClientConfig `json:",inline" mapstructure:",squash"`
+	azclient.AzureAuthConfig `json:",inline" mapstructure:",squash"`
+	*RateLimitConfig         `json:",inline" mapstructure:",squash"`
+	// name of cluster
+	ClusterName string `json:"clusterName,omitempty" mapstructure:"clusterName,omitempty"`
+	// azure resource location
+	Location string `json:"location,omitempty" mapstructure:"location,omitempty"`
+	// subscription ID
+	SubscriptionID string `json:"subscriptionID,omitempty" mapstructure:"subscriptionID,omitempty"`
+	// default resource group where the cluster is deployed
+	ClusterResourceGroup string `json:"clusterResourceGroup,omitempty" mapstructure:"resourceGroup,omitempty"`
+	// name of the virtual network of cluster
+	VnetName string `json:"vnetName,omitempty" mapstructure:"vnetName,omitempty"`
+	// name of the resource group where the virtual network is deployed
+	VnetResourceGroup string `json:"vnetResourceGroup,omitempty" mapstructure:"vnetResourceGroup,omitempty"`
+	// Enable exponential backoff to manage resource request retries
+	CloudProviderBackoff bool `json:"cloudProviderBackoff,omitempty" mapstructure:"cloudProviderBackoff,omitempty"`
+}
+
+// LoadCloudConfigFromFile loads the cloud config from the specified file
+func LoadCloudConfigFromFile(filePath string) (*CloudConfig, error) {
+	if filePath == "" {
+		return nil, fmt.Errorf("failed to load cloud cloudconfig: file path is empty")
+	}
+
+	var config *CloudConfig
+	configReader, err := os.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open cloud cloudconfig file: %w, file path: %s", err, filePath)
+	}
+	defer configReader.Close()
+
+	contents, err := io.ReadAll(configReader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read cloud cloudconfig file: %w, file path: %s", err, filePath)
+	}
+
+	if err := yaml.Unmarshal(contents, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal cloud cloudconfig: %w, file path: %s", err, filePath)
+	}
+
+	if err := config.defaultAndValidate(); err != nil {
```

Comment:
Could you merge my change in PR https://github.com/Azure/fleet-networking/pull/205?